src/V3Ast.cpp

```diff
@@ -99,25 +99,27 @@ AstNode* AstNode::abovep() const {
 string AstNode::encodeName(const string& namein) {
     // Encode signal name raw from parser, then not called again on same signal
     string out;
-    for (string::const_iterator pos = namein.begin(); pos != namein.end(); ++pos) {
-        if ((pos == namein.begin()) ? std::isalpha(pos[0])  // digits can't lead identifiers
-                                    : std::isalnum(pos[0])) {
-            out += pos[0];
-        } else if (pos[0] == '_') {
-            if (pos[1] == '_') {
-                out += "_";
+    const size_t len = namein.length();
+    const char* buf = namein.data();
+    out.reserve(len);
+    for (size_t i = 0; i < len; ++i) {
+        const char ch = buf[i];
+        if (i == 0 ? std::isalpha(ch) : std::isalnum(ch)) {  // digits can't lead identifiers
```

Comment:
It was there before actually so it's likely already covered?

Reply:
It seems either the test is incomplete, or the test would have been failing without your commit?

Reply:
It's ok for Linux but fails on Windows, because regression tests are not run on Windows.

On Linux there's a test for identifiers begin with non-alphas, but it's better to add tests for trailing underscores.